From 6850f4bfb2a38c5ac3cde1bdd936fa330ea6da4e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=89=8B=E7=93=9C=E4=B8=80=E5=8D=81=E9=9B=AA?=
 <nanaeonn@outlook.com>
Date: Thu, 23 Oct 2025 08:55:41 +0800
Subject: [PATCH 08/11] Add SILK audio codec descriptor and update Makefile

Introduces a descriptor for the SILK audio codec in codec_desc.c and updates the silk_v3_decoder Makefile to use CONFIG_SILK_DECODER instead of CONFIG_SILK_V3_DECODER for object file generation.
---
 libavcodec/codec_desc.c             | 7 +++++++
 libavcodec/silk_v3_decoder/Makefile | 4 ++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/libavcodec/codec_desc.c b/libavcodec/codec_desc.c
index d58c3d889c..efdd3f21a1 100644
--- a/libavcodec/codec_desc.c
+++ b/libavcodec/codec_desc.c
@@ -32,6 +32,13 @@
 #define MT(...) (const char *const[]){ __VA_ARGS__, NULL }
 
 static const AVCodecDescriptor codec_descriptors[] = {
+        {
+            .id        = AV_CODEC_ID_SILK,
+            .type      = AVMEDIA_TYPE_AUDIO,
+            .name      = "silk",
+            .long_name = NULL_IF_CONFIG_SMALL("SILK Audio"),
+            .props     = AV_CODEC_PROP_INTRA_ONLY | AV_CODEC_PROP_LOSSY,
+        },
     /* video codecs */
     {
         .id        = AV_CODEC_ID_MPEG1VIDEO,
diff --git a/libavcodec/silk_v3_decoder/Makefile b/libavcodec/silk_v3_decoder/Makefile
index 448fab9141..8c8f53ef33 100644
--- a/libavcodec/silk_v3_decoder/Makefile
+++ b/libavcodec/silk_v3_decoder/Makefile
@@ -1,5 +1,5 @@
 SILK_V3_DECODER_SRCS := $(wildcard $(SRC_PATH)/libavcodec/silk_v3_decoder/src/*.c) \
                         $(wildcard $(SRC_PATH)/libavcodec/silk_v3_decoder/src/*.S)
 
-OBJS-$(CONFIG_SILK_V3_DECODER) += $(SILK_V3_DECODER_SRCS:$(SRC_PATH)/%.c=%.o)
-OBJS-$(CONFIG_SILK_V3_DECODER) += $(SILK_V3_DECODER_SRCS:$(SRC_PATH)/%.S=%.o)
\ No newline at end of file
+OBJS-$(CONFIG_SILK_DECODER) += $(SILK_V3_DECODER_SRCS:$(SRC_PATH)/%.c=%.o)
+OBJS-$(CONFIG_SILK_DECODER) += $(SILK_V3_DECODER_SRCS:$(SRC_PATH)/%.S=%.o)
\ No newline at end of file
-- 
2.47.0.windows.2

